ibm/service/dnsservices/resource_ibm_private_dns_custom_resolver_secondary_zone.go

```diff
@@ -0,0 +1,269 @@
+	d.Set(pdnsSecondaryZoneCreatedOn, resource.CreatedOn)
+	d.Set(pdnsSecondaryZoneModifiedOn, resource.ModifiedOn)
+	d.Set(pdnsSecondaryZoneEnabled, *resource.Enabled)
+
+	return nil
+}
+
+func resourceIBMPrivateDNSSecondaryZoneUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).PrivateDNSClientSession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	idSet := strings.Split(d.Id(), "/")
+	if len(idSet) < 3 {
+		return diag.FromErr(fmt.Errorf("[ERROR] Incorrect ID %s: Id should be a combination of InstanceID/resolverID/secondaryZoneID", d.Id()))
+	}
+	instanceID := idSet[0]
+	resolverID := idSet[1]
+	secondaryZoneID := idSet[2]
+
+	// Check DNS zone is present
+	getZoneOptions := sess.NewGetSecondaryZoneOptions(instanceID, resolverID, secondaryZoneID)
+	_, response, err := sess.GetSecondaryZone(getZoneOptions)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("[ERROR] Error fetching secondary zone:%s\n%s", err, response))
+	}
+
+	// Update DNS zone if attributes has any change
+	if d.HasChange(pdnsSecondaryZoneTransferFrom) ||
+		d.HasChange(pdnsSecondaryZoneDescription) ||
+		d.HasChange(pdnsSecondaryZoneEnabled) {
+		updateSecondaryZoneOptions := sess.NewUpdateSecondaryZoneOptions(instanceID, resolverID, secondaryZoneID)
+		transferFrom := flex.ExpandStringList(d.Get(pdnsSecondaryZoneTransferFrom).([]interface{}))
+		description := d.Get(pdnsSecondaryZoneDescription).(string)
+		enabled := d.Get(pdnsSecondaryZoneEnabled).(bool)
+		updateSecondaryZoneOptions.SetTransferFrom(transferFrom)
+		updateSecondaryZoneOptions.SetDescription(description)
+		updateSecondaryZoneOptions.SetEnabled(enabled)
+
+		mk := "private_dns_secondary_zone_" + instanceID + resolverID
+		conns.IbmMutexKV.Lock(mk)
+		defer conns.IbmMutexKV.Unlock(mk)
+
+		_, response, err := sess.UpdateSecondaryZone(updateSecondaryZoneOptions)
+
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("[ERROR] Error updating DNS Services zone:%s\n%s", err, response))
+		}
+	}
+
+	return resourceIBMPrivateDNSSecondaryZoneRead(ctx, d, meta)
+}
+
+func resourceIBMPrivateDNSSecondaryZoneDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).PrivateDNSClientSession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	idSet := strings.Split(d.Id(), "/")
+	if len(idSet) < 3 {
+		return diag.FromErr(fmt.Errorf("[ERROR] Incorrect ID %s: Id should be a combination of InstanceID/resolverID/secondaryZoneID", d.Id()))
+	}
+	instanceID := idSet[0]
+	resolverID := idSet[1]
+	secondaryZoneID := idSet[2]
+	deleteSecondaryZoneOptions := sess.NewDeleteSecondaryZoneOptions(instanceID, resolverID, secondaryZoneID)
+
+	mk := "private_dns_secondary_zone_" + instanceID + resolverID
+	conns.IbmMutexKV.Lock(mk)
+	defer conns.IbmMutexKV.Unlock(mk)
+	response, err := sess.DeleteSecondaryZone(deleteSecondaryZoneOptions)
+
+	if err != nil {
```

Comment:
Here also you can check 
if response != nil && response.StatusCode == 404 {
			return nil
		}

Reply:
I believe I have addressed this comment